### PR TITLE
vespa-cli: Build endpoints map

### PR DESCRIPTION
**Pull request by hakonhall**
This PR is supposed to be a no-op.  But the idea is to hold on to the full map from cluster name to URL endpoints in cloudTarget, and resue the target across multiple requests using possibly different clusters.

### CODE REVIEW

**Comment by hakonhall** [comment]
FYI @jonmv

File: client/go/vespa/target.go [hunk 1]
```diff
@@ -208,11 +208,34 @@ type cloudTarget struct {
...
 
-	queryURL       string
-	documentURL    string
+	urlsByCluster  map[string]string
 	authConfigPath string
 }
 
+func (t *cloudTarget) resolveEndpoint(cluster string) (string, error) {
+	if cluster == "" {
+		for _, u := range t.urlsByCluster {
```
Comment (mpolden) [inline client/go/vespa/target.go:217]: Isn't this loop unnecessary?

File: client/go/vespa/target.go [hunk 2]
```diff
@@ -208,11 +208,34 @@ type cloudTarget struct {
...
+			}
+		}
+	} else {
+		u := t.urlsByCluster[cluster]
+		if u == "" {
+			clusters := make([]string, len(t.urlsByCluster))
+			for c := range t.urlsByCluster {
+				clusters = append(clusters, c)
+			}
+			return "", fmt.Errorf("unknown cluster '%s': must be one of %v, ", cluster, clusters)
```
Comment (mpolden) [inline client/go/vespa/target.go:231]: Any reason for the trailing `, ` in the message?

File: client/go/vespa/target.go (hunk 1 above)
Comment (hakonhall) [reply client/go/vespa/target.go:217]: How would I get the value of the entry, if the map contains exactly one entry?  I found this which seems ugly:

```
t.urlsByCluster[reflect.ValueOf(t.urlsByCluster).MapKeys()[0]]
```

File: client/go/vespa/target.go (hunk 2 above)
Comment (hakonhall) [reply client/go/vespa/target.go:231]: Thanks, fixed

**Review by mpolden — APPROVED** [review]

**Pull request merged**